src/formatDate.js

```diff
@@ -8,7 +8,39 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
-}
+  const separatorFrom = fromFormat[fromFormat.length - 1];
+  const dateParts = date.split(separatorFrom);
+
+  const dateObj = {};
+
+  fromFormat.slice(0, -1).forEach((part, index) => {
+    dateObj[part] = dateParts[index];
+  });
+
+  const convertYear = (year, format) => {
+    if (format === 'YYYY') {
+      if (year.length === 2) {
+        return parseInt(year) < 30 ? '20' + year : '19' + year;
+      } else {
+        return year;
+      }
+    } else if (format === 'YY') {
+      return year.slice(-2);
+    }
+
+    return year;
+  };
 
+  const newDateParts = toFormat.slice(0, -1).map((part) => {
+    if (part === 'YYYY' || part === 'YY') {
+      return convertYear(dateObj['YYYY'] || dateObj['YY'], part);
+    }
+
+    return dateObj[part];
+  });
```

Comment:
This block of code could be simplified. Instead of checking whether the part is 'YYYY' or 'YY', you could directly pass the part to the `convertYear` function. This would make the code more concise and easier to understand.